controllers/bpfman-agent/internal/test-utils/fake_bpfman_client.go

```diff
@@ -56,7 +56,14 @@ func NewBpfmanClientFakeWithPrograms(programs map[int]*gobpfman.ListResponse_Lis
 }
 
 func (b *BpfmanClientFake) Load(ctx context.Context, in *gobpfman.LoadRequest, opts ...grpc.CallOption) (*gobpfman.LoadResponse, error) {
-	id := rand.Intn(100)
+	var id int
+	for {
+		id = rand.Intn(30000)
+		if _, ok := b.LoadRequests[id]; !ok {
+			break
+		}
+	}
```

Comment:
can u avoid infinite loop and just have maxRetries and fail if its reached ?

Reply:
I figured someone wouldn't like the infinite loop :-).  How about this new solution that doesn't use random numbers?